packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -27,6 +29,16 @@ function ContrastChecker( {
 } ) {
 	const currentBackgroundColor = backgroundColor || fallbackBackgroundColor;
 
+	const { strictContrastCheck } = useSelect(
+		( select ) => ( {
+			strictContrastCheck: select( preferencesStore ).get(
+				'core/edit-post',
```

Comment:
When preferences are saved they're grouped into objects like this:
```js
{
    'core': {
         // preferences shared across editors (there aren't any at the moment)
    },
    'core/edit-post': {
        // preferences that are only for the post editor
    },
    'core/edit-site': {
       // preferences that are only for the site editor
    }
}
```

So you'd need to decide what the right value here should be. A problem is that this is in the 'block-editor' package which is used across both the post and site editor, so having 'core/edit-post' hardcoded doesn't seem like the right option.

I think it would be good to have the `strictContrastCheck` passed in as a prop to the component rather than have the preference built-in to solve that